src/discord.cpp

```diff
@@ -0,0 +1,207 @@
+            mGamesNames[pHost] = name;
+            UpdatePresence();
+            return std::tuple<bool, QString>(true, QString());
+        } else {
+            return std::tuple<bool, QString>(false, QStringLiteral("%1 is not a known game").arg(name));
+        }
+    }
+    return std::tuple<bool, QString>(false, QStringLiteral("Discord integration is not available"));
+}
+
+bool Discord::setArea(Host* pHost, const QString& area)
+{
+    if (mLoaded) {
+        mAreas[pHost] = area;
+        UpdatePresence();
+        return true;
+    }
+    return false;
+}
+
+bool Discord::setCharacterIcon(Host* pHost, const QString& icon)
+{
+    if (mLoaded) {
+        mCharacterIcons[pHost] = icon;
+        UpdatePresence();
+        return true;
+    }
+    return false;
+}
+
+bool Discord::setCharacter(Host* pHost, const QString& text)
+{
+    if (mLoaded) {
+        mCharacters[pHost] = text;
+        UpdatePresence();
+        return true;
+    }
+    return false;
+}
+
+void Discord::timerEvent(QTimerEvent* event)
+{
+    Q_UNUSED(event);
+
+    if (mLoaded) {
+        Discord_RunCallbacks();
+    }
+}
+
+void Discord::handleDiscordReady(const DiscordUser* request)
+{
+    Q_UNUSED(request);
+
+    mudlet::self()->mDiscord.UpdatePresence();
+}
+
+void Discord::handleDiscordDisconnected(int errorCode, const char* message)
+{
+    qWarning() << "Discord disconnected:" << errorCode << message;
+}
+
+void Discord::handleDiscordError(int errorCode, const char* message)
+{
+    qWarning() << "Discord error:" << errorCode << message;
+}
+
+void Discord::handleDiscordJoinGame(const char* joinSecret)
+{
+    Q_UNUSED(joinSecret);
+}
+
+void Discord::handleDiscordSpectateGame(const char* spectateSecret)
+{
+    Q_UNUSED(spectateSecret);
+}
+
+void Discord::handleDiscordJoinRequest(const DiscordUser* request)
+{
+    Q_UNUSED(request);
+}
+
+void Discord::UpdatePresence()
+{
+    DiscordRichPresence discordPresence;
+    memset(&discordPresence, 0, sizeof(discordPresence));
+
+    char buffer[256];
+    buffer[0] = '\0';
+    auto gameName = mGamesNames[mudlet::self()->getActiveHost()].toUtf8();
+    auto gameNameLowercase = mGamesNames[mudlet::self()->getActiveHost()].toLower().toUtf8();
+    auto area = mAreas[mudlet::self()->getActiveHost()].toUtf8();
+    auto characterIcon = mCharacterIcons[mudlet::self()->getActiveHost()].toLower().toUtf8();
+    auto characterText = mCharacters[mudlet::self()->getActiveHost()].toUtf8();
+
+    bool knownGame = mKnownGames.contains(gameNameLowercase);
+
+    if (!gameName.isEmpty()) {
+        sprintf(buffer, "Playing %s", gameName.constData());
+        discordPresence.details = buffer;
+        discordPresence.largeImageKey = gameNameLowercase.constData();
+    }
+
+    if (!area.isEmpty()) {
+        discordPresence.state = area.constData();
+    }
+
+    if (!characterIcon.isEmpty()) {
+        // the game game is unknown, set the small image as the big one so at least something shows
```

Comment:
"game game"

Reply:
You commented twice on the same thing somehow!